casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -73,6 +77,70 @@ class RuntimeManagerImpl[F[_]: Concurrent: Metrics: Span: Log](
   private[this] val computeStateLabel = Metrics.Source(RuntimeManagerMetricsSource, "compute-state")
   private[this] val computeGenesisLabel =
     Metrics.Source(RuntimeManagerMetricsSource, "compute-genesis")
+  private[this] val systemDeployConsumeAllPattern =
+    BindPattern(List(toPar(Expr(EVarBody(EVar(Var(FreeVar(0))))))), freeCount = 1)
+  private[this] val emptyContinuation = TaggedContinuation()
+
+  def playSystemDeploy[S <: SystemDeploy, Env0](runtime: Runtime[F], systemDeploy: S)(
+      normalizerEnv: NormalizerEnv[Env0]
+  )(
+      implicit provided: NormalizerEnv.Provides[Env0, systemDeploy.Env],
+      ev: ToEnvMap[systemDeploy.Env]
+  ): F[Either[SystemDeployFailure, SystemDeployResult[systemDeploy.Result]]] = {
+    implicit val c: _cost[F] = runtime.cost
+    for {
+      softCheckpoint <- runtime.space.createSoftCheckpoint()
+      result <- EitherT
+                 .liftF(
+                   Interpreter[F]
```

Comment:
This needs to be a `Interpreter[F].injAttempt` call, which means we should pull `c: _cost[F]` into the `liftF` scope. That's also where we'll have to provide the `rand` implicit.

Reply:
`into the liftF scope` ?

Reply:
The minimal scope. 

```
EitherT.liftF {
   implicit c: _cost[F] = runtime.cost
   Interpreter[F].injAttempt(...)
}
```

Reply:
Fixed in 7ca418a